File: Casks/github-copilot-app.rb
cask "github-copilot-app" do
  version "1.1.15"
  # 上游 github/app 按架构分包：core 的 url/sha 是 darwin-arm64 的，本 tap 取
  # GitHub-Copilot-darwin-x64.dmg（主二进制 file 实测 x86_64 thin）。
  # Intel 包的 sha256 由检查器下载实算（core 那份属于另一个架构的包，不能用）。
  # 镜像到本仓 Release（tag github-copilot-app-<ver>，资产名沿用上游 basename，
  # 无空格无需换点）：本 tap 政策是所有 cask 都镜像最新版（AGENTS 11.34）；
  # url 用 #{version} 插值过 audit 的 unversioned 检查。版本判据走 core cask
  # API（api/cask/github-copilot-app.json 的 .version）；上游另有 floating
  # 短链 gh.io/copilot-app-mac-intel（releases/latest/download/...）——HEAD
  # 探测不跟随跳转，检查器只能用版本化直链。
  sha256 "008500d3d1a8f080419133ab7d76e0d18660d15b221c15f613cc7adcb7b6326a"

  url "https://github.com/Bemly/homebrew-tahoe/releases/download/github-copilot-app-#{version}/GitHub-Copilot-darwin-x64.dmg"
  name "GitHub Copilot"
  desc "Native client for GitHub Copilot"
  homepage "https://github.com/github/app"

  auto_updates true
  # 本 tap 只收录 Intel(x86_64) + macOS 26(Tahoe) 及以上可用的二进制。
  depends_on arch: :x86_64
  depends_on macos: :tahoe

  app "GitHub Copilot.app"

  zap trash: [
    "~/Library/Application Support/com.github.githubapp",
    "~/Library/Caches/com.github.githubapp",
    "~/Library/Preferences/com.github.githubapp.plist",
    "~/Library/WebKit/com.github.githubapp",
  ]
end

File: updater/github-copilot-app.swift
// updater/github-copilot-app.swift —— 与 Casks/github-copilot-app.rb 一一同名的检查器入口
//
// GitHub Copilot 桌面 app（github/app）。在 homebrew/core 里是 **cask**，
// core 那份是 darwin-arm64 的 dmg（url/sha 都是 arm64 的，不能直接取）；
// 本 tap 取上游按架构分包的 GitHub-Copilot-darwin-x64.dmg。
//
// 版本判据：formulae.brew.sh 的 cask API（api/cask/github-copilot-app.json
// 的 .version），走 UpdaterCore 的 brewCask 分支。
// 下载源必须用版本化直链（releases/download/v<ver>/...）：上游还有 floating
// 链接 gh.io/copilot-app-mac-intel（releases/latest/download/...）——核心的
// HEAD 探测不跟随跳转，floating 链接的 302/直链形态不可用。
// sha256：core 那份属于 arm64 包 → 置 nil，核心回退下载实算（~268MB，
// 仅新版本时发生）。
// 镜像：uploadRelease=true（全 cask 镜像政策，AGENTS 11.34）——有更新时上传
// 本仓 Release（tag github-copilot-app-<ver>，资产名沿用上游 basename，
// 无空格无需换点；旧 release 自动清理），url 整条改写为 Release 地址。
//
// 注意：Swift 要求实参顺序与 CheckConfig.init 的形参声明一致。

import Foundation

@main
struct GithubCopilotAppCheck {
    static func main() {
        runCheck(CheckConfig(
            formula: "github-copilot-app",
            formulaPath: "Casks",
            isCask: true,
            brewName: "github-copilot-app",
            downloadURL: { version in
                "https://github.com/github/app/releases/download/v\(version)/GitHub-Copilot-darwin-x64.dmg"
            },
            brewCask: true,
            uploadRelease: true
        ))
    }
}
